src/main/java/org/isf/medicalstock/service/MedicalStockIoOperations.java

```diff
@@ -366,6 +359,9 @@ public List<Movement> getMovements(String wardId, LocalDateTime dateFrom, LocalD
 		for (int i = 0; i < pMovementCode.size(); i++) {
 			Integer code = pMovementCode.get(i);
 			Movement movement = movRepository.findById(code).orElse(null);
+			if (movement == null) {
+				throw new OHServiceException(new OHExceptionMessage("Movement '" + code + "' not found."));
+			}
```

Comment:
Why is this needed?

Reply:
Because `findBy()` returns `Optional`so that is why `orElse(null)` is added.   Thus if the `code` is not found the object is `null`.  That can happen if the movement got deleted since the `find` in line 357 was executed.   It is just protecting against a very rare NPE.

Reply:
I agree upon very rare... in theory the `findMovementWhereDatesAndId` could return a `List<Movement>` directly and the for cycle could be just remoed. I don't remember why it is returing a `List<Integer>` 🤔

Also here, we could put just a reminder and handle it in another issue.

Reply:
Added TODO comment

Reply:
Thanks!